ebean-core/src/main/java/io/ebeaninternal/server/querydefn/DefaultOrmQuery.java

```diff
@@ -1232,10 +1232,8 @@ public boolean isForceHitDatabase() {
 
   @Override
   public void resetBeanCacheAutoMode(boolean findOne) {
-    if (useBeanCache == CacheMode.AUTO) {
-      if (!findOne || useQueryCache != CacheMode.OFF) {
-        useBeanCache = CacheMode.OFF;
-      }
+    if (useBeanCache == CacheMode.AUTO && useQueryCache != CacheMode.OFF) {
+      useBeanCache = CacheMode.OFF;
     }
   }
```

Comment:
Effectively removing the `!findOne` part.

Note that only SpiQuery.Mode.LAZYLOAD_BEAN and Mode.NORMAL queries come in here with CacheMode.AUTO now (due to explicitly setting cache mode OFF everywhere else).